Add Xbox triplet doc page (#75)
microsoft/vcpkg-docs: vcpkg/users/platforms/xbox.md

---

> title: Xbox
> description: Use x64 community triplets for Xbox development.
> ms.date: 04/27/2023

Reviewer: ```suggestion
title: Xbox
description: Use x64 community triplets for Xbox development.
author: walbourn
ms.author: viromer
ms.date: 8/8/2023
```

Feel free to change `ms.author` to your MS alias if you want to be notified about changes to this file.

---
# Xbox

> [!NOTE]

> > XBox is not tested as part of vcpkg repository's CI process, so regressions can occur as part of library updates. PRs improving support are welcome!

Reviewer 2: Should be "Xbox"

There are two variants of each triplet, selecting between static and dynamic linking.

| Console          | vcpkg community triplets     |
|------------------|------------------------------|
| Xbox One         | x64-xbox-xboxone             |
|                  | x64-xbox-xboxone-static      |
| Xbox Series X\|S | x64-xbox-scarlett            |
|                  | x64-xbox-scarlett-static     |

The ``static`` linking triplets are set to use the MSVC Runtime as a DLL (i.e. ``VCPKG_CRT_LINKAGE dynamic``) as that's the recommended and supported usage scenario.

## Building with the Windows SDK

The Windows SDK (19041) or later includes the system headers required to build against the ``WINAPI_FAMILY_GAMES`` API partition used by the Xbox Game OS. Therefore, most libraries can be built against these triplets without having access to the NDA *Microsoft GDK with Xbox Extensions*.

## Building with the Microsoft GDK with Xbox Extensions

> Libraries that make use of DirectX graphics APIs must be built using a custom variant of Direct3D 12.X used by the Xbox Game OS, and this reuqires access to the NDA *Microsoft GDK with Xbox Extensions*. Note that the Xbox Game OS doe snot support Direct3D 11, Direct2D/DirectWrite, OpenGL, GDI, or any older version of Direct3D.

Reviewer 2: "requires" typo

Reviewer 2: Also "does not"

> For libraries that require these headers, they make use of **ms-gdkx** "stub" port to indicate this requirement. This port looks for paths pointed to by ``GameDKLatest``, ``GameGXDKLatest``, and ``GameGRDKLatest`` environment variables.

Reviewer: ```suggestion
For libraries that require these headers, they make use of **`ms-gdkx`** "stub" port to indicate this requirement. This port looks for paths pointed to by ``GameDKLatest``, ``GameGXDKLatest``, and ``GameGRDKLatest`` environment variables.
```

Traditionally, we use code-style on triplet names and port names.

> > Note that Xbox developers are responsible for ensuring they and their publisher have appropriate and correct licensing terms for use of any 3rd party software components per the terms of their Title License Agreement. Be sure to seek appropriate legal advice for any use of open-source components in your title.

Reviewer: ```suggestion
> [!NOTE] 
> Xbox game developers are responsible for ensuring that them and their publishers have appropriate and correct licensing terms for use of any 3rd party software components per the terms of their Title License Agreement. Be sure to seek appropriate legal advice for any use of open-source components in your title.
```

Using `[!NOTE]` to make it pop up.

## UWP on Xbox

> If building for [UWP on Xbox](/windows/uwp/xbox-apps/getting-started), make use of the existing **x64-uwp** triplet instead.

Reviewer: ```suggestion
If building for [UWP on Xbox](/windows/uwp/xbox-apps/getting-started), make use of the existing `x64-uwp` triplet instead.
```

Traditionally, we use code-style on triplet names and port names.

## Maintainer notes

> CMake projects for these triplets are built using ``CMAKE_SYSTEM_NAME`` set to "Windows". To indicate the Xbox console being targeted, the `XBOX_CONSOLE_TARGET` variable is set to "xboxone" or "scarlett".

Reviewer: ```suggestion
CMake projects for these triplets are built using ``CMAKE_SYSTEM_NAME`` set to "Windows". To indicate the Xbox console being targeted, the `XBOX_CONSOLE_TARGET` variable is set to "`xboxone`" or "`scarlett`".
```

Using code-style to avoid possibly unwanted machine-translation to other languages.

---
title: Xbox
description: Use x64 community triplets for Xbox development.
author: walbourn
ms.author: viromer
ms.date: 8/8/2023
---
# Xbox

> [!NOTE]

> > Xbox is not tested as part of vcpkg repository's CI process, so regressions can occur as part of library updates. PRs improving support are welcome!

Reviewer 3: ```suggestion
> Xbox is not tested as part of vcpkg repository's continuous integration process, so regressions can occur as part of library updates. 
```

## Xbox community triplets

> vcpkg includes [community triplets](https://github.com/microsoft/vcpkg/tree/master/triplets/community) for [Microsoft GDK](https://aka.ms/gdk) PC and [Xbox](https://aka.ms/gdkx) development. They make use of the Visual C++ compiler toolset (VS 2019 16.11 or later).

Reviewer 3: ```suggestion
vcpkg includes [community triplets](https://github.com/microsoft/vcpkg/tree/master/triplets/community) for [Microsoft GDK](https://aka.ms/gdk) PC and [Xbox](https://aka.ms/gdkx) development. They make use of the MSVC compiler toolset (Visual Studio 2019 version 16.11 or later).
```

> There are two variants of each triplet, selecting between static and dynamic linking.

Reviewer 3: ```suggestion
There are two variants for each triplet: static and dynamic linking.
```

| Console          | vcpkg community triplets     |
|------------------|------------------------------|
| Xbox One         | x64-xbox-xboxone             |
|                  | x64-xbox-xboxone-static      |
| Xbox Series X\|S | x64-xbox-scarlett            |
|                  | x64-xbox-scarlett-static     |

The ``static`` linking triplets are set to use the MSVC Runtime as a DLL (i.e. ``VCPKG_CRT_LINKAGE dynamic``) as that's the recommended and supported usage scenario.

## Building with the Windows SDK

The Windows SDK (19041) or later includes the system headers required to build against the ``WINAPI_FAMILY_GAMES`` API partition used by the Xbox Game OS. Therefore, most libraries can be built against these triplets without having access to the NDA *Microsoft GDK with Xbox Extensions*.

## Building with the Microsoft GDK with Xbox Extensions

Libraries that make use of DirectX graphics APIs must be built using a custom variant of Direct3D 12.X used by the Xbox Game OS, and this requires access to the NDA *Microsoft GDK with Xbox Extensions*. Note that the Xbox Game OS does not support Direct3D 11, Direct2D/DirectWrite, OpenGL, GDI, or any older version of Direct3D.

> For libraries that require these headers, they make use of the **`ms-gdkx`** "stub" port to indicate this requirement. This port looks for paths pointed to by ``GameDKLatest``, ``GameGXDKLatest``, and ``GameGRDKLatest`` environment variables.

Reviewer 3: ```suggestion
Libraries that require these headers must use the **`ms-gdkx`** "stub" port to indicate this requirement. This port looks for paths pointed to by ``GameDKLatest``, ``GameGXDKLatest``, and ``GameGRDKLatest`` environment variables.
```
